validator/client/src/main/java/tech/pegasys/teku/validator/client/BeaconProposerPreparer.java

```diff
@@ -111,6 +119,27 @@ public Optional<Eth1Address> getFeeRecipient(final BLSPublicKey publicKey) {
         .orElse(defaultFeeRecipient);
   }
 
+  // Cannot set a fee recipient if the key is specified in the configuration file
+  // Cannot set fee recipient to 0x00
+  public void setFeeRecipient(final BLSPublicKey publicKey, final Eth1Address eth1Address)
+      throws SetFeeRecipientException {
+    if (eth1Address.equals(Eth1Address.ZERO)) {
+      throw new IllegalArgumentException("Cannot set fee recipient to 0x00 address.");
+    }
```

Comment:
Why can't we set it to 0x00?

Reply:
the set api suggested 'default address' was not valid which is 0x00 from discussions on channel, and with you... should i not check that?

Reply:
I guess I can see the argument for it as it's probably the user making a mistake.  Just seems slightly odd to filter out this one user mistake.  If the spec says it's not allowed I'm fine with it though.